media/server/gstplayer/source/GstCapabilities.cpp

```diff
@@ -109,6 +119,42 @@ bool GstCapabilities::isMimeTypeSupported(const std::string &mimeType)
     return m_supportedMimeTypes.find(mimeType) != m_supportedMimeTypes.end();
 }
 
+bool GstCapabilities::doesSinkOrDecoderHaveProperty(MediaSourceType mediaType, const std::string &propertyName)
+{
+    // Get all element factories
+    GstElementFactoryListType factoryListType{GST_ELEMENT_FACTORY_TYPE_SINK | GST_ELEMENT_FACTORY_TYPE_DECODER};
```

Comment:
So my understanding is that we only look for elements that are GST_ELEMENT_FACTORY_TYPE_SINK   AND GST_ELEMENT_FACTORY_TYPE_DECODER, is that correct? Should we be looking for SINKs     OR     DECODERs?